validator_client/src/initialized_validators.rs

```diff
@@ -248,37 +249,40 @@ impl InitializedValidator {
             } => {
                 let signing_url = build_web3_signer_url(&url, &def.voting_public_key)
                     .map_err(|e| Error::InvalidWeb3SignerUrl(e.to_string()))?;
+
                 let request_timeout = request_timeout_ms
                     .map(Duration::from_millis)
                     .unwrap_or(DEFAULT_REMOTE_SIGNER_REQUEST_TIMEOUT);
 
-                let builder = Client::builder().timeout(request_timeout);
-
-                let builder = if let Some(path) = root_certificate_path {
-                    let certificate = load_pem_certificate(path)?;
-                    builder.add_root_certificate(certificate)
+                // Check if a client has already been initialized for this remote signer url.
```

Comment:
This is a bit frustrating, but I think if we want to support different values for `root_certificate_path`, `request_timeout_ms`, `client_identity_path` and `client_identity_password` then we need to key the hashmap on all of them... Otherwise another signer using the same URL is liable to accidentally use a `Client` with the wrong parameters. I feel like 99% of the time in practice it would be fine, but we should probably support the edge case of different parameters for the same URL.

Maybe the map could be keyed by `SigningDefinition`? Or a type nested inside ` SigningDefinition::Web3Signer` like `SigningDefinition::Web3Signer(Web3SignerDefinition { /* fields */ })`.

Let me know what you think

Reply:
Yep, good call!